lib/API/JsonApi/v1/Resource/SuiteBeanResource.php

```diff
@@ -431,9 +440,147 @@ private function fromResource(Resource $resource)
      * @param Relationship $relationship
      * @return SuiteBeanResource
      */
-    public function withRelationship(\SuiteCRM\API\JsonApi\v1\Resource\Relationship $relationship) {
+    public function withRelationship(\SuiteCRM\API\JsonApi\v1\Resource\Relationship $relationship)
+    {
         $relationshipName = $relationship->getRelationshipName();
         $this->relationships[$relationshipName]['data'] = $relationship->toJsonApiResponse();
         return clone $this;
     }
+
+    /**
+     * Set the attributes to download a base64 encoded file
+     * @param \File $bean
```

Comment:
It seems $bean could be Document as well.

Reply:
I have modified the both Note and Document to extends File. So file is the most appropriate choice here.

Reply:
You are expecting Document and Note!
@param \Document|\Note $bean

Reply:
@daniel-samson I think the problem is: When you extends Note from File then the Note also instance of File. Also If you force the type to File "kind" of class in the function definition part then below unnecessary to test it in the method body line 465 (`} elseif ($bean instanceof \File) {`) because when the php interpreter read this it's already a File or child of a File, ensue also pointless to throwing an exception in the last `else` at line 467, 468.  
More simple if you just use a struct like this:
```
if ($bean instanceof \Document) {
    $this->retrieveFileFromDocumentBean($bean, $fieldName);
} else {
    $this->retrieveFileFromFileBean($bean, $fieldName);
}
```
Question is it still will works properly?